### DISCUSSION TITLE
Export/publish combined RSS Feed

### DISCUSSION THREAD

**Discussion by VincenzoKoestler** [FreshRSS/FreshRSS#6654] [Q&A] [answered]
Hey Community, from what I have read it should be possible to publish an feed as rss feed externally but I cannot seem to find the query functions. How can I publish an rss feed that is a collection of all the feeds I maintain in freshRSS? 

What do I want to achieve? I want to use freshRSS to scrape newspages for news and create my own RSS feed (kind of as an alternative to rss.app). Is that possible?
Thanks so much for your help! :)

**Comment by Alkarex** [comment, marked as answer]
You can share any FreshRSS view (e.g. feed, category, user label, or search) by RSS or HTML
https://freshrss.github.io/FreshRSS/en/users/user_queries.html#share-your-user-queries
But watch out not to infringe copyright if you share publicly

**Reply by VincenzoKoestler** [reply]
Hey Alkarex, I read about the queries but I cannot seem to figure out how to do it. I have read the link you sent but I do not have the option to save the current search as bookmark. 

Edit: nevermind, the explanation is outdated, I found it; its the bookmark sign. I dont know how I did not manage before!
![image](https://github.com/user-attachments/assets/69ddf0fb-f876-4019-8ae9-e1ba02846a4e)


Also for further readers encountering the same issue:
its also possible to share the whole rss feed via: `/?a=rss` (can be controlled via `&nb=` to define the number of items and `&hours=` to define the time frame.

Thanks a lot Alkarex!

**Comment by VincenzoKoestler** [comment]
@Alkarex Hey Alexandre, one further question, is it possible to create such a feed with JSON output as well or is JSON only integrated for fetching so far? :)

**Reply by Alkarex** [reply]
I have just implemented it. Check https://github.com/FreshRSS/FreshRSS/pull/6655 and tests welcome

**Reply by VincenzoKoestler** [reply]
Amazing!! I am running on docker, how can I test it out?

**Reply by Alkarex** [reply]
Assuming you are using our official Docker image, then:

To test a PR, you need to point your Docker or Docker Compose to the right PR, as such:

```yml
  freshrss:
    image: freshrss/freshrss:6655
    build:
      context: https://github.com/Alkarex/FreshRSS.git#share-by-greader-format
      dockerfile: Docker/Dockerfile-Alpine
    ...
```

* https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/freshrss/docker-compose.yml
* https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#build-custom-docker-image

And by the way, PR welcome to add some tips [to our documentation](https://github.com/FreshRSS/FreshRSS/blob/edge/docs/en/contributing.md) on how to test patches

**Reply by VincenzoKoestler** [reply]
Thanks Alexandre, I will try that :)
I am still heavily in the process of learning a lot and not that familiar with how Github works as well but if possible will add what I learn to the documentation :)

One last question (hopefully):
To learn programming I decided to build myself my own personal dashboard application with react. Due to limited resources I have decided to try to achieve this serverless. Now I have been able to fetch news feeds using rss2json but some pages just cannot be fetched which is where I decided to get involved with FreshRSS. However, when I try to fetch my new RSS Feed with rss2json I get an CORS error. Now, hopefully, I can move away from using 3rd party services like rss2json but for my understanding, why am I encountering the CORS error when trying to fetch my RSS feed but not when trying to fetch newspaper rss feeds?

**Reply by Alkarex** [reply]
CORS is mostly governed by HTTP headers (which you can inspect, and also modify if you are the Web site owner).
Check https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

**Reply by VincenzoKoestler** [reply]
> CORS is mostly governed by HTTP headers (which you can inspect, and also modify if you are the Web site owner). Check https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

yes but it seems like the rss/json feed does not allow to be accessed due to CORS, can that be? I cannot fetch the json output of FreshRSS clientside to use it in my app, I get rejected right away, the feed does not seem to send the "Access-Control-Allow-Origin" header?

**Reply by Alkarex** [reply]
This is typically something that is decided by the server admin (i.e. you) at your Web server / proxy level.
But we could consider adding some default rules for the `/api/` path

**Reply by Alkarex** [reply]
https://github.com/FreshRSS/FreshRSS/pull/6655 merged in `edge` branch for additional testing

**Reply by VincenzoKoestler** [reply]
> This is typically something that is decided by the server admin (i.e. you) at your Web server / proxy level.

But with simple rules it would be possible to work with the feeds without having to use a proxy inbetween...

> But we could consider adding some default rules for the `/api/` path

That would be absolutely amazing!!



Regarding the gReader, I tried out the json export yesterday and it works great! I saw the menu in the main screen was a bit messed up in that release but I will check again on the latest :Edge branch

**Reply by Alkarex** [reply]
> without having to use a proxy inbetween

But you most likely already have a Web server / proxy already, for instance for the HTTPS part, don't you?

Feel free to open a new issue so we can remember it.

> I saw the menu in the main screen was a bit messed up

If you spot any problem, make sure to report them in a new issue with some screenshots

**Reply by VincenzoKoestler** [reply]
not really, I use cloudflare tunneling to publish the services

will do!, I need to update the the latest version first though :)

**Reply by Alkarex** [reply]
I have made a patch https://github.com/FreshRSS/FreshRSS/pull/6659 Please test.
But even with Cloudflare, you are able to customize the HTTP headers: https://stackoverflow.com/questions/76983353/how-to-configure-cors-of-access-control-allow-origin-using-cloudflare

**Reply by VincenzoKoestler** [reply]
you are amazingly fast, thanks so much!! I will install the update and get back to you :)
